reframe/core/meta.py

```diff
@@ -401,6 +401,26 @@ def __getattr__(cls, name):
             f'class {cls.__qualname__!r} has no attribute {name!r}'
         ) from None
 
+    def setvar(cls, name, value):
+        '''Set the value of a variable (unless the value is a descriptor).'''
+
+        var_space = super().__getattribute__('_rfm_var_space')
+        if name in var_space:
+            if not hasattr(value, '__get__'):
+                var_space[name].define(value)
+                return
+            elif var_space[name].field is not value:
+                desc = '.'.join([cls.__qualname__, name])
+                raise ReframeSyntaxError(
+                    f'cannot override variable descriptor {desc!r}'
+                )
+            else:
+                # We are injecting the underlying descriptor of the variable,
+                # so set it as a normal class attribute
+                super(cls).__setattr__(name, value)
+        else:
+            raise AttributeError(f'{cls.__name__!r} has no variable {name!r}')
```

Comment:
This is what I meant with my comment above.
```suggestion
        try:
            var_space = super().__getattribute__('_rfm_var_space')
            if name in var_space:
                if not hasattr(value, '__get__'):
                    var_space[name].define(value)
                    return True
                elif var_space[name].field is not value:
                    desc = '.'.join([cls.__qualname__, name])
                    raise ReframeSyntaxError(
                        f'cannot override variable descriptor {desc!r}'
                    )
        except AttributeError:
            '''Catch early access attempt to the variable space.'''
        finally:
            return False

```

Reply:
Then, the part in `__setattr__` is just
```python
if cls.setvar(name, value):
    return
```
and you don't have to suppress that exception in the loader when setting the variables from the command line

Reply:
One problem with this suggestion is that we can't return in `finally` and also raise an exception in some other code path. The exception will be suppressed. If we stick with `setvar()` returning boolean I can fix that.

Reply:
I guess we could also just move that `return False` after the `try/except`, but this also looks good to me.